argo/src/Argo/ServerState.hs

```diff
@@ -0,0 +1,180 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
```

Comment:
This file seems the most critical to document for future maintainers, since it's where the handling of caching and mutable state happens, and that's the trickiest to reason about. I think it would be useful for future-you-and-me-and-others to add some explanatory comments for the fields of `ServerState` and top-level docs for the various functions in this module.

Reply:
Indeed - I agree completely and will do so.